### PR TITLE
Remove the rds-aurora terraform module

### PR DESCRIPTION
**Pull request by Kamsiy**
## Description

TLDR: This PR removes the [rds-aurora module](https://github.com/terraform-aws-modules/terraform-aws-rds-aurora/tree/v2.22.0) and defines the configurations for the database resources.

Artemis uses v2.22.0 of the [`terraform-aws-rds-aurora`](https://github.com/terraform-aws-modules/terraform-aws-rds-aurora/tree/v2.22.0) module. The latest version of this module is v9.9.1. 

Updating to the latest version of this module will require rewriting the Terraform configurations. Also, version 2.22.0 restricts our ability to make specific updates to the Terraform configurations for AWS Aurora.

To avoid future blockers, this PR removes this module and defines the AWS resources needed for Artemis

## How Has This Been Tested?

tested in a dev environment

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation change

## Checklist

- [x] My code follows conforms to the coding standards.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: backend/terraform/modules/analyzer/db.tf [hunk 1]
```diff
@@ -64,6 +53,70 @@ resource "aws_rds_cluster_parameter_group" "cluster_parameter_group" {
...
+
+############################ 
+# Aurora Cluster Instance ##
+############################
+resource "aws_rds_cluster_instance" "cluster_instance" {
+  count = 1 # Create 1 db instance
+
+  apply_immediately          = true
+  auto_minor_version_upgrade = true
+  availability_zone          = element(var.database_availability_zones, length(var.database_availability_zones) - 1)
```
Comment (ZoogieZork) [inline backend/terraform/modules/analyzer/db.tf:90]: To support more than a single instance (i.e., `count` > 1), we'll want to spread the instances across the available AZs.

I'm assuming we're picking the last element in the list to match what's currently deployed?  If so, we could do something like:

```terraform
availability_zone = element(reverse(var.database_availability_zones), count.index)
```

File: backend/terraform/modules/analyzer/db.tf [hunk 2]
```diff
@@ -1,37 +1,25 @@
...
-  engine                          = "aurora-postgresql"
-  engine_version                  = var.db_engine_version
-  auto_minor_version_upgrade      = true
+#######################
+# Aurora RDS Cluster ##
+#######################
+resource "aws_rds_cluster" "aurora" {
   apply_immediately               = true
-  db_parameter_group_name         = aws_db_parameter_group.db_parameter_group.name
+  availability_zones              = ["us-east-2a", "us-east-2b", "us-east-2c"]
```
Comment (ZoogieZork) [inline backend/terraform/modules/analyzer/db.tf:6]: Should this be `var.database_availability_zones`?

**Review by ZoogieZork — APPROVED** [review]
